app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala

```diff
@@ -23,19 +42,32 @@ class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost) {
   }
 
   def parseResponse(xmlString: String) : Option[AgentDetails] = {
-    val XML = scala.xml.XML.withSAXParser(SAXParserFactory.newInstance().newSAXParser())
-
-    val xml = XML.loadString(xmlString)
+    val xml: Elem = toXmlElement(xmlString)
     val agentDetails = xml \ "AllocatedAgents" \ "AgentDetails"
-    val agentCode = (agentDetails \ "AgentCode").head.text
+    if (agentDetails.nonEmpty) {
+      val agentCode = (agentDetails \ "AgentCode").head.text
 
-    val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
-      AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
+        AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      }
+      Some(AgentDetails(agentCode, credentials))
```

Comment:
We only don't know the business POV, all the others suggest it's a list:

XSD is `<xs:element maxOccurs="unbounded" name="AgentDetails" type="AgentDetailsType" />`.

Alos, GG doc quite unambiguously claims that it returns a list of agent codes, check part 3.3.1.27.